cmd/celestia/main.go

```diff
@@ -29,8 +31,18 @@ func init() {
 	rootCmd.SetHelpCommand(&cobra.Command{})
 }
 
+func shutdownTracer() {
+	tp, ok := otel.GetTracerProvider().(*tracesdk.TracerProvider)
+	if ok {
+		// Flush and shutdown traces to capture all spans.
+		_ = tp.ForceFlush(context.Background())
+		_ = tp.Shutdown(context.Background())
+	}
+}
```

Comment:
This logic ideally should live next to the logic configuring the tracer, `cmd/flags_misc.go`. Alternatively, we can move this to FX, as you've mentioned, but we will also need to move the _setup_ logic to the FX as we do for metrics(`nodebuilder/settings.go`)

Reply:
We can consider similar `WithTraces` option

Reply:
@Wondertan see latest change using FX lifecycle.